src/python/turicreate/test/test_text_analytics.py

```diff
@@ -255,6 +255,15 @@ def test_tf_idf(self):
         assert len(empty_dict_sf) == 1
         assert len(empty_dict_sf.apply(lambda x: len(x) == 0)) == 1
 
+    def test_count_words_dict_type(self):
+        sa = tc.SArray([{'alice bob': 1, 'Bob alice': 0.5},
+                            {'a dog': 0, 'a dog cat': 5}])
+        result = tc.text_analytics.count_words(sa)
+        expected = [{'bob': 1.5, 'alice': 1.5}, {'a': 5.0, 'dog': 5.0, 'cat': 5.0}]
```

Comment:
For each dictionary in `expected` the values are all the same. So this unit test is only really checking one possible code path. When testing it is best to cover as many possible code path combinations as possible. 

Try setting `sa` to something like:
```
tc.SArray([{'Alice bob mike': 1, 'Bob Alice Sue': 0.5, '': 100},
                             {'a dog cow': 0, 'a dog cat': 5, 'mice dog': -1}])
```
Then figure out what `expected` should be and see if this test passes.

Reply:
for word count, the count number shouldn't be negative.

Reply:
@Jarvi-Izana - that's a very good point. You're quite right.

Negative counts and the empty string as a key would probably be good error/edge case tests but if you're only adding only one test it probably doesn't make sense to add either of these.